cirq/ops/common_channels.py

```diff
@@ -31,109 +31,196 @@
 class AsymmetricDepolarizingChannel(gate_features.SingleQubitGate):
     """A channel that depolarizes asymmetrically along different directions."""
 
-    def __init__(self, p_x: float, p_y: float, p_z: float) -> None:
+    def __init__(self,
+                 p_x: Optional[float] = None,
+                 p_y: Optional[float] = None,
+                 p_z: Optional[float] = None,
+                 num_qubits: int = 1,
+                 error_probabilities: Optional[Dict[str, float]] = None
+                ) -> None:
         r"""The asymmetric depolarizing channel.
 
-        This channel applies one of four disjoint possibilities: nothing (the
-        identity channel) or one of the three pauli gates. The disjoint
-        probabilities of the three gates are p_x, p_y, and p_z and the
-        identity is done with probability 1 - p_x - p_y - p_z. The supplied
-        probabilities must be valid probabilities and the sum p_x + p_y + p_z
-        must be a valid probability or else this constructor will raise a
-        ValueError.
+        This channel applies one of 4**n disjoint possibilities: nothing (the
+        identity channel) or one of the 4**n pauli gates. The disjoint
+        probabilities of the 4**n gates.
 
         This channel evolves a density matrix via
 
             $$
-            \rho \rightarrow (1 - p_x - p_y - p_z) \rho
-                    + p_x X \rho X + p_y Y \rho Y + p_z Z \rho Z
+            \sum_i p_i Pi \rho Pi
             $$
 
+        Where i varies from 0 to 4**n-1, representing the 4**n Pauli gates,
+        represented by Pi. The input \rho is the density matrix before the
+        depolarization.
+
         Args:
             p_x: The probability that a Pauli X and no other gate occurs.
             p_y: The probability that a Pauli Y and no other gate occurs.
             p_z: The probability that a Pauli Z and no other gate occurs.
+            num_qubits: Number of qubits
+            error_probabilities: Dictionary of string (Pauli operator) to its
+                probability
```

Comment:
nit: maybe add an example, so that it's clear to the user what they're supposed to pass in without having to read the code below